fe1-web/src/core/redux/ServerReducer.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
+
+import { PublicKey, Server, ServerAddress, ServerState } from 'core/objects';
+
+/**
+ * Reducer & associated function implementation to store all known servers
+ */
+
+export interface ServerReducerState {
+  byAddress: Record<ServerAddress, ServerState>;
+  allAddresses: ServerAddress[];
+}
+
+const initialState: ServerReducerState = {
+  byAddress: {},
+  allAddresses: [],
+};
+
+export const SERVER_REDUCER_PATH = 'servers';
+
+const serverSlice = createSlice({
+  name: SERVER_REDUCER_PATH,
+  initialState,
+  reducers: {
+    addServer: (state: Draft<ServerReducerState>, action: PayloadAction<ServerState>) => {
+      const server = action.payload;
+
+      state.byAddress[server.address] = server;
+      state.allAddresses.push(server.address);
+    },
+
+    updateServer: (state: Draft<ServerReducerState>, action: PayloadAction<ServerState>) => {
+      const updatedServer = action.payload;
+
+      if (!(updatedServer.address in state.byAddress)) {
+        return;
+      }
+
+      state.byAddress[updatedServer.address] = updatedServer;
+    },
+
+    removeServer: (state, action: PayloadAction<string>) => {
+      const serverAddress = action.payload;
+
+      if (serverAddress in state.byAddress) {
+        delete state.byAddress[serverAddress];
+        state.allAddresses = state.allAddresses.filter((id) => id !== serverAddress);
+      }
+    },
+
+    clearAllServers: (state) => {
+      state.byAddress = {};
+      state.allAddresses = [];
+    },
+  },
+});
+
+export const { addServer, clearAllServers, removeServer, updateServer } = serverSlice.actions;
+
+export const serverReduce = serverSlice.reducer;
+
+export default {
+  [SERVER_REDUCER_PATH]: serverSlice.reducer,
+};
+
+export const getServerState = (state: any): ServerReducerState => state[SERVER_REDUCER_PATH];
+
+/**
+ * A function to directly retrieve the public key from the redux store.
+ * NOTE: This function does not memoize the result. If you need this, use makeServerSelector instead
```

Comment:
should this be a `@remarks` field ?

Reply:
Seems like it, thank you! :) Apart from `@param` and `@returns?  I do not know the tsdoc tags very well 😅